coresimd/x86/i686/rdrand.rs

```diff
@@ -0,0 +1,80 @@
+//! RDRAND and RDSEED instructions for returning random numbers from an Intel
+//! on-chip hardware random number generator which has been seeded by an on-chip
+//! entropy source.
+extern "platform-intrinsic" {
+    fn x86_rdrand16_step() -> (u16, i32);
+    fn x86_rdrand32_step() -> (u32, i32);
+    fn x86_rdrand64_step() -> (u64, i32);
+    fn x86_rdseed16_step() -> (u16, i32);
+    fn x86_rdseed32_step() -> (u32, i32);
+    fn x86_rdseed64_step() -> (u64, i32);
+}
```

Comment:
We don't use `platform-intrinsic` in this crate. Can you link to the LLVM intrinsics directly?

Reply:
No, as was discussed in the issue LLVM  type {I<width>, i32} can not be currently represented by Rust.

Reply:
I'm not sure that means we want to start using platform-intrinsics here.

We should probably wait for @alexcrichton to review this and help decide how to proceed.

Reply:
He [suggested](https://github.com/rust-lang-nursery/stdsimd/issues/325#issuecomment-366839315) to use `asm!` block as a temporary solution, so I think platform-intrinsics will be an ok temporary solution as well.

Reply:
Ah yeah the platform intrinsic ABI here is the way to go for now. This may change in the future, but we can always just update the crate!

In any case this PR looks good to me, just waiting on the rust-lang/rust PR!